exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/QueryManager.java

```diff
@@ -338,7 +345,8 @@ private QueryProfile getQueryProfile(UserException ex) {
         .setStart(startTime)
         .setEnd(endTime)
         .setTotalFragments(fragmentDataSet.size())
-        .setFinishedFragments(finishedFragments.get());
+        .setFinishedFragments(finishedFragments.get())
+        .addAllOptions(queryOptions);
```

Comment:
We [serialize](https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/planner/fragment/SimpleParallelizer.java#L292) and [deserialize](https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentContext.java#L149) options elsewhere. We should do something similar here as well.